src/main/java/carpet/network/CarpetClient.java

```diff
@@ -16,6 +16,7 @@
     public static final int HI = 69;
     public static final int HELLO = 420;
     public static final int DATA = 1;
+    public static final int STRUCTURE_BLOCK = 2;
```

Comment:
I am suggesting not creating new types, just hook into DATA - there is nothing special about structure blocks here - data handles all sorts of communication, for rules, scarpet etc.